src/commands/staff/warn.ts

```diff
@@ -37,36 +54,23 @@ export default class extends Command {
 			}
 		}
 
-		target.author.send(`Your message was deleted in ${target.channel} by ${msg.author.tag}. Below is the given reason:\n${reason}`)
+		target.author.send(`Your message was deleted in ${target.channel} by ${interaction.user.tag}. Below is the given reason:\n${reason}`)
```

Comment:
If the DM send fails it might be worth creating a priv thread between the user and the bot regarding the warning.

Reply:
Is there a need? They will be sent an email.

Reply:
Fair point